### PR TITLE
use select widget to render oneOf / anyOf control

### LINKED ISSUE #1219: allow custom widget for oneOf / anyOf select
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [ ] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://mozilla-services.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

When providing a custom select widget, I expected it to be used for the oneOf / anyOf select. The MultiSchemaField component just renders a `select` instead of looking up the select widget ([link](https://github.com/mozilla-services/react-jsonschema-form/blob/d9bcf1acfa3fd15814654129054b3c65501f5271/src/components/fields/MultiSchemaField.js#L160)). 

### Steps to Reproduce

1. Create a schema with a oneOf / anyOf property
2. Create a custom select widget and pass to the form: `<Form widgets={{ SelectWidget: MyCustomSelect }} />`
3. Observe that the select used for the oneOf / anyOf does not render the custom widget.

#### Expected behavior

I would expect that my custom widget would be rendered instead.

#### Actual behavior

A `select` element is used.

### Version

1.3.0

### PR DESCRIPTION
**Pull request by a-b-r-o-w-n**
### Reasons for making this change

Fixes #1219. oneOf / anyOf will now use the `SelectWidget` to render the top control. This can be customized by the user. 

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by a-b-r-o-w-n** [comment]
I have 3 tests failing locally on master, as well as this branch. Also, I am curious if any other props need to be plumbed through to the select widget. It feels a little weird to provide a stubbed schema, but that may be ok for this use case. Just let me know if I can update anything.

File: src/components/fields/MultiSchemaField.js [hunk 1]
```diff
@@ -141,7 +141,9 @@ class AnyOfField extends Component {
     } = this.props;
 
     const _SchemaField = registry.fields.SchemaField;
+    const { widgets } = registry;
     const { selectedOption } = this.state;
+    const Widget = getWidget({ type: "string" }, "select", widgets);
```
Comment (epicfaace) [inline src/components/fields/MultiSchemaField.js]:
```suggestion
    const Widget = getWidget({ type: "number" }, "select", widgets);
```

**Comment by epicfaace** [comment]
> A stubbed schema seems fine, thanks!
> 
> You should probably pass in the [props listed here](https://react-jsonschema-form.readthedocs.io/en/latest/advanced-customization/#custom-widget-components), as well.

In addition to the autofocus and placeholder attributes (see #1223 -- maybe we can get these attributes from the uiSchema entry of the oneOf once https://github.com/mozilla-services/react-jsonschema-form/issues/1141 is resolved). For now, just add which attributes you are able to without needing the uiSchema.

**Comment by a-b-r-o-w-n** [comment]
@epicfaace Let me know if we shouldn't just get the uiOptions now without #1141. It didn't appear to make a difference since the defaults are falsey.

**Comment by a-b-r-o-w-n** [comment]
@epicfaace are you still waiting for changes?

**Comment by epicfaace** [comment]
@a-b-r-o-w-n sorry for the delay. Can you add tests for oneOf & anyOf in which you actually define a `SelectWidget` in the `widgets` prop for the form?

**Comment by a-b-r-o-w-n** [comment]
🤦‍♂️ I thought I already had. I'll update the specs

**Comment by epicfaace** [comment]
Thanks!

**Comment by mravenash** [comment]
Can anyone add an example on the play ground for this?  It would be really helpful.
Regards
Avenash

**Comment by a-b-r-o-w-n** [comment]
@mravenash I don't think the playground supports custom widgets since it is just json. Here is a codesandbox link that has an example though: https://codesandbox.io/embed/6wxv10mz2k

**Comment by mravenash** [comment]
> @mravenash I don't think the playground supports custom widgets since it is just json. Here is a codesandbox link that has an example though: https://codesandbox.io/embed/6wxv10mz2k

For the above schema in the sanbox, I am able to override the custom text widget, but doesn't work for SelectWidget. At what point does the SelectWidget function called?
Any comments will be helpful.

`const schema = {
  type: 'object',
  oneOf: [
    {
      title: 'Person',
      properties: {
        name: {
          type: 'string',
          title: 'Name:',
        },
        birthday: {
          type: 'string',
          title: 'Birthday:',
        },
      },
      required: ['name', 'birthday'],
    },
    {
      title: 'Dog',
      properties: {
        name: {
          type: 'string',
          title: 'Name:',
        },
        breed: {
          type: 'string',
          title: 'Breed:',
        },
      },
      required: ['name', 'breed'],
    },
  ],
};

function TextWidget(props) {
  return (<TextField label="Default TextField" {...props} />);
}

function SelectWidget(props) {
  return (<Dropdown
    source={['Person', 'Dog']}
    {...props}
  />);
}
const widgets = {
  TextWidget,
  SelectWidget
};
const CreateForm = ({ classes }) => (
  <Panel>
    <ColumnLayout />
    <div className={classes.formPadding} >
      <JsonSchemaForm
        schema={schema}
        widgets={widgets}
        showErrorList={false}
      />
    </div>
  </Panel>
);
`

**Comment by katepma** [comment]
Is there any way to remove the border that encapsulates the custom widgets? I see this css classname  **fieldset** and its border-width is set to 2px. Is there any way to override it? Any inputs will be appreciated.
Regards
Avenash

**Comment by a-b-r-o-w-n** [comment]
What I have done is to either provide a custom [field template](https://react-jsonschema-form.readthedocs.io/en/latest/advanced-customization/#object-field-template) or just use css to modify the border:
```css
.MyForm {
  fieldset {
    border: none;
  }
}
```

**Pull request merged**